app/src/main/java/com/pvp/app/service/FriendServiceImpl.kt

```diff
@@ -56,16 +56,39 @@ class FriendServiceImpl @Inject constructor(
 
     override suspend fun remove(email: String) {
         database
-            .runTransaction { transaction ->
-                val document = database
-                    .collection(identifier)
-                    .document(email)
+            .collection(identifier)
+            .document(email)
+            .delete()
+            .await()
 
-                transaction.delete(document)
-            }
+        database
```

Comment:
When removing friend object, we are removing the initial object and then iterating through **every** other friend object, no matter if they are related or not, then updating each. At least it seems so.

This is very redundant iteration, when we could take the initial object and check what friends it stores, then map to those specific friends in iteration.